### ISSUE TITLE
Bug with excluded variables

### ISSUE THREAD

**Issue by isomemo** [Pandora-IsoMemo/bmsc-app#11]
@arunge @mgross 

I used this data: 
[test.xlsx](https://github.com/Pandora-IsoMemo/bmsc-app/files/9758082/test.xlsx)

Here you can see the inputs:

![image](https://user-images.githubusercontent.com/46485054/195149866-a23caa15-c96b-4326-b112-6bfeb60cf814.png)

The model has an interaction depth of 2 but I wanted to exclude all interactions except for these three:

Source1*C1 AND Source2*C2 AND Source3*C3

You can see the above here:

![image](https://user-images.githubusercontent.com/46485054/195150241-73b52b36-64e2-4b19-a30a-040202ac2f2a.png)

![image](https://user-images.githubusercontent.com/46485054/195150314-0f6fe92f-75de-4adb-aaf3-8899ab89dc8d.png)

![image](https://user-images.githubusercontent.com/46485054/195150456-f21b8dfc-3a93-4b7c-928b-82524f1de94e.png)

However, some of the excluded interactions ended up being included in the model output (e.g. Source1*C2 or C3*Source2):

![image](https://user-images.githubusercontent.com/46485054/195150722-6de7de9a-b468-4bbe-8808-b98fa0503e30.png)

**Comment by mgross** [comment]
@isomemo 

This is certainly a bug, thanks for reporting. After investigating, I found out that it was a reordering bug that occured in an internal function of R. I worked around this and this should work with 22.10.1

Apart from that, looking at your data the model doesn't make sense, actually. C1, C2, and C3 are constant, such that neither the inclusion of C1,C2 or C3 or the inclusion of C1:Source1,.. make any sense. (this is due to perfect multicollinearity, which will only lead to numerical instability).
